src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,44 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todo: Todo | null;
+  setModalOpen: (val: boolean) => void;
```

Comment:
The `setModalOpen` prop is expected to be a function that takes a boolean, but it is used to close the modal without passing any argument. This could lead to confusion. Consider renaming it to something like `closeModal` for clarity, or ensure it is used consistently with its intended purpose.